File: testing-common/src/main/java/io/opentelemetry/instrumentation/testing/util/TestLatestDeps.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.testing.util;

public final class TestLatestDeps {

  private static final boolean TEST_LATEST_DEPS = Boolean.getBoolean("testLatestDeps");

  /**
   * Returns {@code true} when the test is being run with {@code -PtestLatestDeps=true}, i.e.
   * against the latest published versions of the instrumented library instead of the pinned
   * earliest-supported versions.
   */
  public static boolean testLatestDeps() {
    return TEST_LATEST_DEPS;
  }

  private TestLatestDeps() {}
}

File: conventions/src/main/kotlin/io/opentelemetry/instrumentation/gradle/TestLatestDeps.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.gradle

import org.gradle.api.Project

/** Returns true when the build is run with `-PtestLatestDeps=true`. */
val Project.testLatestDeps: Boolean
  get() = gradle.startParameter.projectProperties["testLatestDeps"] == "true"
